### ISSUE TITLE
DockerAccessWithHcClient.inspectContainer throws DockerAccessException

### ISSUE THREAD

**Issue by danielwegener** [fabric8io/docker-maven-plugin#566]
In Version 0.16.3:

`DockerAccessWithHcClient.inspectContainer` never returns `null` if a container can not be found (as suggested by the javadoc of `DockerAccess`) but rather throws an `DockerAccessException(e, "Unable to retrieve container name for [%s])`. This is harmful when `io/fabric8/maven/docker/util/StartOrderResolver.java:127`  tries to look for existing externally linked containers via `io.fabric8.maven.docker.service.QueryService#hasContainer` (It ends up in `Cannot resolve image dependencies for start order` messages) whenever the start order resolver checks for external containers.

`DockerAccessWithHcClient.inspecContainer` should rather expect 404s and return `null` if encountered as in

```java
String response = delegate.get(url, HTTP_OK, HTTP_NOT_FOUND);
return (response == null)? null : new ContainerDetails(new JSONObject(response));
```

and the `ApacheHttpClientDelegate` should maybe translate `404` responses into `null`s.

Unfortunately I see no configuration workaround to get around this

**Comment by rhuss** [comment]
I consider this a bug and will make the change you suggest. All other methods also have the same semantics that null means 'not-found'.

**Comment by martinborisov87** [comment]
Hi, I still have a problem with that. 
I have a docker:stop in my "pre-integration-test" phase in order to stop containers just in case they are still running.
Before this fix I was getting "Unable to retrieve container name for [%s]" if the container was not existing, but now I am getting a really strange one - "Failed to execute goal io.fabric8:docker-maven-plugin:0.16.4:stop (stop-before-test) on project: Execution stop-before-test of goal io.fabric8:docker-maven-plugin:0.16.4:stop failed: A JSONObject text must begin with '{' at 1 [character
 2 line 1] -> [Help 1]"

I didn't have any problems with the previous plugin versions. I decided to update the version, because I needed to use the "network" feature, introduced later.

Thanks
